src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Stream.cs

```diff
@@ -12,6 +13,14 @@ namespace System.Text.Json
 {
     public static partial class JsonSerializer
     {
+        // We flush the Stream when the buffer is >=90% of capacity.
+        // This threshold is a compromise between buffer utilization and minimizing cases where the buffer
+        // needs to be expanded\doubled because it is not large enough to write the current property or element.
+        // We check for flush after each object property and array element is written to the buffer.
```

Comment:
` // We check for flush after each object property and array element is written to the buffer.` 

Should we also mention dictionary values?

Reply:
I'll qualify with "JSON property" instead of just property (dictionary values are normally properties). Also FWIW these comments were preexisting - this PR just moved from a method to the class in order to re-use the const.